Init Fault Report Log. Fixes: #130

SukiCZ opened the pull request:
This creates a `FaultReportLog` model and creates logs when `FaultReport` is being changed. 

Fixes: #130

berk76 commented:
Thank you @SukiCZ it looks good. Will you also create fault log page accessible from particular fault detail?

berk76 commented:
Please merge from master and re-make your migrations.

berk76 reviewed (changes requested):
I am testing it and I am not sure choosed approach is right.

1. When I create new fault no entry in log is created - I would expect entry of type `created`
2. When I modify Fault  `modified` entry is created but has different time from `modified` field in Fault. 
3. New `Modified` field in fault is unnecessary and confusing because this information is in log and it is not clear which time contains.
4. No entry is created when I use shortcuts through following functions (because they are not using Form):
    * `views_faults.faults_fault_take_ticket_view()`
    * `views_faults.faults_fault_close_ticket_view()`

SukiCZ commented:
Sorry, I've missed the part with separate "fault log page" 📄, fixed
🗃️ Migrations rebased


1. I assumed `FaultReportLog.TYPE_CREATED` was unnecessary because of `FaultReport.created`, but added
1. `modified` had different value in FaultReport (FR) and FaultReportLog (FRL), because updating FR was happening before creatiion of FRL. Same Effect will apply to creation and logging of creation. (`FR.modified` was removed)
1. Django [DateField](https://docs.djangoproject.com/en/dev/ref/models/fields/#datefield) has two options: `auto_now_add` = (when the object is first created) and `auto_now` = (every time the object is saved). It corresponds to `created/modified`
1. Thanks. I've added logging to those views :ok_hand:

berk76 commented:
Thank you for your changes @SukiCZ I have tested it quickly and it look good.

1. I would insist on `Created` entry because creator may or may not assign somebody during ticket creation step and this information is missing
2. Please show full name of user and resolver - login can be quite cryptic
3. Show blank field instead None
4. Please add link `Back to ticket`

Anyway I will need more time to test this change properly.

berk76 commented:
BTW : I like type checking you introduced in managers

SukiCZ commented:
Thanks for your feedback.

> I like type checking you introduced in managers

You might like [mypy](http://pypi.org/p/mypy)
